### ISSUE TITLE
[Bug]: constrained_layout_pads as kwargs of Figure

### ISSUE THREAD

**Issue by StefRe** [matplotlib/matplotlib#22029]
### Bug summary

According to the [figure documentation (other parameters)](https://matplotlib.org/stable/api/figure_api.html#module-matplotlib.figure) it should be possible pass `figure` kwargs in `plt.subplots()`:
```
fig, ax = plt.subplots(layout='constrained',
                       constrained_layout_pads={'w_pad': 0.5, 'h_pad': 0.5})
```

which results, however, in a

`TypeError: set_constrained_layout_pads() takes 1 positional argument but 2 were given`

The documentation mentions a single value (`float`), although the parameter name is in plural (and [set_constrained_layout_pads](https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure.set_constrained_layout_pads) takes up to four parameters):

> **constrained_layout_pads**    float, default: rcParams["figure.constrained_layout.w_pad"] (default: 0.04167)

but `plt.subplots(constrained_layout_pads=0.5)` results in the same error.

### Code for reproduction

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots(layout='constrained',
                       constrained_layout_pads={'w_pad': 0.5, 'h_pad': 0.5})
```

### Actual outcome

```

  File "/usr/lib/python3.10/site-packages/matplotlib/pyplot.py", line 1434, in subplots
    fig = figure(**fig_kw)

  File "/usr/lib/python3.10/site-packages/matplotlib/pyplot.py", line 787, in figure
    manager = new_figure_manager(

  File "/usr/lib/python3.10/site-packages/matplotlib/pyplot.py", line 306, in new_figure_manager
    return _backend_mod.new_figure_manager(*args, **kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 3493, in new_figure_manager
    fig = fig_cls(*args, **kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/figure.py", line 2238, in __init__
    super().__init__(**kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/figure.py", line 216, in __init__
    self.set(**kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/artist.py", line 116, in <lambda>
    cls.set = lambda self, **kwargs: Artist.set(self, **kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/artist.py", line 1164, in set
    return self.update(kwargs)

  File "/usr/lib/python3.10/site-packages/matplotlib/artist.py", line 1066, in update
    ret.append(func(v))

TypeError: Figure.set_constrained_layout_pads() takes 1 positional argument but 2 were given
```

### Expected outcome

no error

### Additional information

also discussed at [discours](https://discourse.matplotlib.org/t/setting-constrained-layout-pads-in-plt-subplots/22482)

### Operating system

Arch

### Matplotlib Version

3.5.1

### Matplotlib Backend

Qt5Agg

### Python version

3.10.1

### Installation

Linux package manager

**Comment by jklymak** [comment]
Thanks for opening this here.  See also https://github.com/matplotlib/matplotlib/pull/20115#issuecomment-999013693

**Comment by tacaswell** [comment]
xref https://github.com/matplotlib/matplotlib/pull/20115#issuecomment-998975855

Matplotlib `Artist` instances have many properties (approximately the set given by the getter/setters (and we have looked into moving to properties / traitlets / ... for more managed access to them and there are some edge cases like this one that make it hard).  As a nice short hand we have a convention that usually `**kwargs` to the `__init__` method of the `Artist` sub-classes get tossed into `self.update` which is  https://github.com/matplotlib/matplotlib/blob/b57506eb00afa4b09cda023e4d76ed06ca976217/lib/matplotlib/artist.py#L1053-L1077

Thus we can write things like

```python
ax.plot(x, y, linewidth=5, color='green')
```

rather than

```python
ln, = ax.plot(x, y)
ln.set_linewidth(5)
ln.set_color('green')
```

This works on the expectation that a) for a keyword `foo` there is a method `set_foo` on the instance b) the method will take 1 positional argument (the value of the keyword) that can be passed in and "do the right thing" (that some of the `set_foo` methods violate this and take optional arguments is actually one of the biggest sticking points in moving to property-like usage).

One fix would be to exclude `constrained_layout_pads` from functions that are allowed to be passed to via `kwargs` (so you would get a clearer exception), but given that we are going to be py38+ for our next release, we have access to positional-only arguements so we can change the signature to 

```python
    def set_constrained_layout_pads(self, d=None, /, *,
                                    w_pad=None, h_pad=None,
                                    wspace=None, hspace=None):
```

which along with a bit of munging in body can make this work as @StefRe expected.

**Comment by jklymak** [comment]
Right, the unexpected thing for me was that this turned up in the docs - obviously we never meant this to be a kwarg for `subplots`, even if someone could have called it if they knew what to look for.  I guess we could have easily made `set_constrained_layout_pads` something that didn't start with `set`.  

Anyhow, I'm fine w/ the going forward 3.8+ plan.  It'd be great to accept a dict as a kwarg.  I'm a little concerned that we need to do something for MPL 3.5.x about the documentation.  Can we just define `FigureBase.set` locally, filter out `constrained_layout_pads` for now, and remember to revert it for 3.6?

**Comment by tacaswell** [comment]
I think if we go with @anntzer 's suggestion of filtering on `bind(None)` rather than parameter count, then we can keep that in place going forward _and_ add the positional-only to make this work for mpl36.

**Comment by timhoffm** [comment]
This will be easier to handle once we define artist properties explicitly. Xref proposal in #22749
